### PR TITLE
CSS module: fonts

### LINKED ISSUE #176: CSS Module: Fonts
Page: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_fonts

 - [x] brief (5-paragraph approx) description
 - [ ] demonstration example
 - [x] properties
 - [x] descriptors
 - [x] data types
 - [x] associated properties
 - [ ] glossary terms
 - [x] https://drafts.csswg.org/css-fonts/

### PR DESCRIPTION
**Pull request by estelle**
fixes https://github.com/openwebdocs/project/issues/176

### CODE REVIEW

**Comment by RTLY059** [comment]
> возникает [openwebdocs/project#176](https://github.com/openwebdocs/project/issues/176)

File: files/en-us/web/css/css_fonts/index.md [hunk 1]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
-- {{cssxref("line-height")}}
 
-### At-rules
-
-- {{cssxref("@font-face")}}
-- {{cssxref("@font-feature-values")}}
+### At-rules and descriptors
+
+{{cssxref("@font-face")}}
+
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:53]:
```suggestion
At-rule: {{cssxref("@font-face")}}
Descriptors:
```

File: files/en-us/web/css/css_fonts/index.md [hunk 2]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+- {{cssxref("@font-face/font-family", "font-family")}}
+- {{cssxref("@font-face/font-feature-settings", "font-feature-settings")}}
+- {{cssxref("@font-face/font-language-override", "font-language-override")}}
+- {{cssxref("@font-face/font-named-instance", "font-named-instance")}}
+- {{cssxref("@font-face/font-stretch", "font-stretch")}}
+- {{cssxref("@font-face/font-style", "font-style")}}
+- {{cssxref("@font-face/font-variation-settings", "font-variation-settings")}}
+- {{cssxref("@font-face/font-weight", "font-weight")}}
+- {{cssxref("@font-face/line-gap-override", "line-gap-override")}}
+- {{cssxref("@font-face/src", "src")}}
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:66]:
```suggestion
- {{cssxref("@font-face/size-adjust", "size-adjust")}}
- {{cssxref("@font-face/src", "src")}}
```
"size-adjust" is not in the spec but we have a page on this descriptor

File: files/en-us/web/css/css_fonts/index.md [hunk 3]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+- {{cssxref("@font-face/unicode-range", "unicode-range")}}
+
+{{cssxref("@font-feature-values")}}
+
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:70]:
```suggestion
At-rule: {{cssxref("@font-feature-values")}}
Descriptor:
```

File: files/en-us/web/css/css_fonts/index.md [hunk 4]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+- {{cssxref("@font-feature-values/font-display", "font-display")}}
+
+{{cssxref("@font-palette-values")}}
+
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:74]:
```suggestion
At-rule: {{cssxref("@font-palette-values")}}
Descriptors:
```

File: files/en-us/web/css/css_fonts/index.md [hunk 5]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+- [`<font-features-tech>`](/en-US/docs/Web/CSS/@supports#font-tech)
+- [`<font-variant-css2>`](/en-US/docs/Web/CSS/font-variant)
+- [`<font-format>`](/en-US/docs/Web/CSS/@supports#font-format)
+- [`<font-stretch-css3>`](/en-US/docs/Web/CSS/font-stretch#values)
+- [`<font-tech>`](/en-US/docs/Web/CSS/@supports#font-tech)
+- [`<numeric-figure-values>`](/en-US/docs/Web/CSS/font-variant-numeric#values)
+- [`<numeric-fraction-values>`](/en-US/docs/Web/CSS/font-variant-numeric#values)
+- [`<numeric-spacing-values>`](/en-US/docs/Web/CSS/font-variant-numeric#values)
+- [`<font-weight-absolute>`](/en-US/docs/Web/CSS/font-weight#values)
+- [`<historical-lig-values>`](/en-US/docs/Web/CSS/font-variant-ligatures#values)
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:100]: `<historical-lig-values>` can be moved up with other `font-variant-ligatures` types

Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:100]: Can we group them like so (the unbulleted items will be rendered with paragraph spacing):

```
- {{cssxref("absolute-size")}}
- {{cssxref("generic-family")}}
- {{cssxref("relative-size")}}

`font-feature-settings` type:
- [`<feature-tag-value>`](/en-US/docs/Web/CSS/font-feature-settings#values)

`font-format` type:
- [`<font-format>`](/en-US/docs/Web/CSS/@supports#font-format)

`font-stretch` type:
- [`<font-stretch-css3>`](/en-US/docs/Web/CSS/font-stretch#values)

`font-tech` types:
- [`<color-font-tech>`](/en-US/docs/Web/CSS/@supports#font-tech)
- [`<font-features-tech>`](/en-US/docs/Web/CSS/@supports#font-tech)
- [`<font-tech>`](/en-US/docs/Web/CSS/@supports#font-tech)

`font-variant` types:
- [`<font-variant-css2>`](/en-US/docs/Web/CSS/font-variant)
- [`<east-asian-variant-values>`](/en-US/docs/Web/CSS/font-variant#values)
- [`<east-asian-width-values>`](/en-US/docs/Web/CSS/font-variant#values)

`font-variant-ligatures` types:
- [`<common-lig-values>`](/en-US/docs/Web/CSS/font-variant-ligatures#values)
- [`<contextual-alt-values>`](/en-US/docs/Web/CSS/font-variant-ligatures#values)
- [`<discretionary-lig-values>`](/en-US/docs/Web/CSS/font-variant-ligatures#values)
- [`<historical-lig-values>`](/en-US/docs/Web/CSS/font-variant-ligatures#values)

`font-variant-numeric` types:
- [`<numeric-figure-values>`](/en-US/docs/Web/CSS/font-variant-numeric#values)
- [`<numeric-fraction-values>`](/en-US/docs/Web/CSS/font-variant-numeric#values)
- [`<numeric-spacing-values>`](/en-US/docs/Web/CSS/font-variant-numeric#values)

`font-weight` type:
- [`<font-weight-absolute>`](/en-US/docs/Web/CSS/font-weight#values)

```

File: files/en-us/web/css/css_fonts/index.md [hunk 6]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+- {{domxref("CSSFontFeatureValuesRule")}}
+- {{domxref("CSSFontPaletteValuesRule")}}
 
 ## Guides
 
-- [Fundamental text and font styling](/en-US/docs/Learn/CSS/Styling_text/Fundamentals)
-  - : In this beginner's learning article we go through all the basic fundamentals of text/font styling in detail, including setting font weight, family and style, font shorthand, text alignment and other effects, and line and letter spacing.
+- [Learn: Fundamental text and font styling](/en-US/docs/Learn/CSS/Styling_text/Fundamentals)
+
+  - : This beginner's learning article covers the basic fundamentals of text/font styling, including setting font weight, family and style, the {{cssxref("font")}} shorthand, text alignment and other effects, and line and letter spacing.
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:112]:
```suggestion
  - : This beginner's learning article covers the basic fundamentals of text and font styling. It covers how to set the font weight, family, and style by using the {{cssxref("font")}} shorthand and how to align text and manage line and letter spacing.
```

File: files/en-us/web/css/css_fonts/index.md [hunk 7]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+
+- [Learn: Web fonts](/en-US/docs/Learn/CSS/Styling_text/Web_fonts)
+
+  - : This beginner's learning article explains how to use custom fonts with your web page to allow for more varied, custom text styling.
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:116]:
```suggestion
  - : This beginner's learning article explains how to use custom fonts on your web page to allow for more varied and custom text styling.
```

File: files/en-us/web/css/css_fonts/index.md [hunk 8]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+
 - [Variable fonts guide](/en-US/docs/Web/CSS/CSS_fonts/Variable_fonts_guide)
-  - : **Variable fonts** are an evolution of the OpenType font specification that enables many different variations of a typeface to be incorporated into a single file, rather than having a separate font file for every width, weight, or style. This article will give you all you need to know to get you started using variable fonts.
+
+  - : **Variable fonts** are an OpenType font specification that enables many different variations of a typeface to be incorporated into a single file, rather than having a separate font file for every width, weight, or style. This article will get you started using variable fonts.
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:124]:
```suggestion
  - : This article will help you get started with using variable fonts.
```
We don't need to define variable fonts here (we have done so in the intro paras). Keeping only the description of the article. Moved the "OpenType" info to the intro para explaining variable fonts.

File: files/en-us/web/css/css_fonts/index.md [hunk 9]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+
+- [Improving font perfomance](/en-US/docs/Learn/Performance/CSS#improving_font_performance)
+
+  - : Part of the CSS perfomance guide discussing font loading, loading only needed glyphs, and defining font display behavior with the `font-display` descriptor.
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:128]:
```suggestion
  - : This article, part of the CSS performance guide, discusses font loading, loading only the required glyphs, and defining font display behavior with the `font-display` descriptor.
```

File: files/en-us/web/css/css_fonts/index.md [hunk 10]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+
+## Related concepts
+
+- CSS {{cssxref("line-height")}} property
+- CSS {{cssxref("text-transform")}} property
+- CSS {{cssxref("letter-spacing")}} property
```
Comment (dipikabh) [inline files/en-us/web/css/css_fonts/index.md:134]:
```suggestion
- {{cssxref("letter-spacing")}} CSS property
- {{cssxref("line-height")}} CSS property
- {{cssxref("text-transform")}} CSS property
```
- Making alphabetic and keeping the unlinked text together

(Just got an idea - maybe next to each of these, we can mention the spec/module they're part of. We would need to do this for all "Related concepts" sections in other modules.)

**Review by dipikabh — CHANGES_REQUESTED** [review]
Thank you, @estelle, for adding all the missing pieces (at-rules, related descriptors, data types, interface!). This is really great info 🙌.

File: files/en-us/web/css/css_fonts/index.md [hunk 11]
```diff
@@ -7,82 +7,139 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
 
-## Basic example
+A font is a resource file containing the visual representation of characters, mapping character codes to glyphs representing letters, numbers, punctuation and even emojis of a typeface. A font family is a group of fonts that share common design styles and font properties with each member of the group providing different ways of displaying the gylphs, varying by stroke weight, slant or relative width, among others. A font generally contains is a single style of a typeface, such as Helvetica that is Bold and Italic; a font family is the complete set of styles. Including a such a font is done by defining a separate `@font-face` declaration for each font resource.
 
-The following example shows a simple use of basic font properties to style a paragraph of text.
+The properties, at-rules, and descriptors of the CSS fonts module enable downloading multiple variations of a font while defining which font file for a particular font characteristic, along with fallback instructions should a resource not load. The CSS font selection mechanism describes how to match a given set of CSS font properties to a single font face.
```
[the remark this answers is not in the document]
Comment (RTLY059) [reply files/en-us/web/css/css_fonts/index.md:14]: Least a reply

File: files/en-us/web/css/css_fonts/index.md [hunk 12]
```diff
@@ -7,82 +7,140 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
 
-### At-rules
-
-- {{cssxref("@font-face")}}
-- {{cssxref("@font-feature-values")}}
+### At-rules and descriptors
+
+At-rule: {{cssxref("@font-face")}}
+Descriptors:
+- {{cssxref("@font-face/ascent-override", "ascent-override")}}
```
[the remark this answers is not in the document]
Comment (RTLY059) [reply files/en-us/web/css/css_fonts/index.md]: esable

[the remark this answers is not in the document]
Comment (RTLY059) [reply files/en-us/web/css/css_fonts/index.md]: > **[mdn-linter]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
> ```suggestion
> 
> - {{cssxref("@font-face/ascent-override", "ascent-override")}}
> ```
> 
esable

File: files/en-us/web/css/css_fonts/index.md [hunk 13]
```diff
@@ -7,82 +7,140 @@ spec-urls: https://drafts.csswg.org/css-fonts/
...
+- {{cssxref("@font-face/font-variation-settings", "font-variation-settings")}}
+- {{cssxref("@font-face/font-weight", "font-weight")}}
+- {{cssxref("@font-face/line-gap-override", "line-gap-override")}}
+- {{cssxref("@font-face/size-adjust", "size-adjust")}}
+- {{cssxref("@font-face/src", "src")}}
+- {{cssxref("@font-face/unicode-range", "unicode-range")}}
+
+At-rule: {{cssxref("@font-feature-values")}}
+Descriptor:
+- {{cssxref("@font-feature-values/font-display", "font-display")}}
```
[the remark this answers is not in the document]
Comment (RTLY059) [reply files/en-us/web/css/css_fonts/index.md]: > **[mdn-linter]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
> ```suggestion
> 
> - {{cssxref("@font-feature-values/font-display", "font-display")}}
> ```
> 

estenlle

File: files/en-us/web/css/css_fonts/index.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (RTLY059) [reply files/en-us/web/css/css_fonts/index.md]: > **[mdn-linter]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
> ```suggestion
> 
> - {{cssxref("@font-face/ascent-override", "ascent-override")}}
> ```
> 
esablle

[the remark this answers is not in the document]
Comment (RTLY059) [reply files/en-us/web/css/css_fonts/index.md]: > **[mdn-linter]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
> ```suggestion
> 
> - {{cssxref("@font-face/ascent-override", "ascent-override")}}
> ```
> 

establle

[the remark this answers is not in the document]
Comment (RTLY059) [reply files/en-us/web/css/css_fonts/index.md]: > **[mdn-linter]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
> ```suggestion
> 
> - {{cssxref("@font-face/ascent-override", "ascent-override")}}
> ```
> 

eatablle

**Review by dipikabh — APPROVED** [review]
This is looking great! Thanks for the updates.

(maybe in a future PR, we can move the description of OpenType font features (currently in the "Guides" section describing the "OpenType font features guide") to the intro - seems like good info to be available along with other font info and not confined to the "Guides" section)

**Pull request merged**